staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/interface.go

```diff
@@ -0,0 +1,97 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/cel"
+
+	"github.com/google/cel-go/common/types/ref"
+)
+
+type FailurePolicy string
+
+const (
+	Fail   FailurePolicy = "Fail"
+	Ignore FailurePolicy = "Ignore"
+)
+
+type EvaluatorFunc func(a admission.Attributes, params *unstructured.Unstructured) []PolicyDecision
+
+// Dependency Injected into the PolicyDefinition's `Compile` function to assist
+// with converting types and values to/from CEL-typed values
+type ObjectConverter interface {
+	// Given a GVR, look up the openapi or JSONSchemaProps, structural schema, etc.
+	// and compile it into something that can be used to turn objects into CEL
+	// values
+	DeclForResource(gvr schema.GroupVersionResource) *cel.DeclType
+
+	// Object may be a typed native object or an unstructured object
+	ValueForObject(value runtime.Object, decl *cel.DeclType) *ref.Val
+}
+
+// An interface for internal policy binding type. Implemented by mock/testing types,
+// and to be implemented by the public API types once they have completed API review.
+//
+// The interface closely mirrors the format and functionality of the
+// PolicyDefinition proposed in the KEP.
+type PolicyDefinition interface {
+	runtime.Object
+
+	// returns namespace/name
+	GetNamespace() string
+	GetName() string
+
+	// says whether this policy definition matches the provided admission resource request
+	Matches(a admission.Attributes) bool
+
+	Compile(
+		// Definition is provided with a converter which may be used by the
+		// return evaluator function to convert objects into CEL-typed objects
+		objectConverter ObjectConverter,
+	) (EvaluatorFunc, error)
+
+	// GVK for the CRD used as the source of parameters used in the evaluation
+	// of instances of this policy
+	GetParamSource() schema.GroupVersionKind
+
+	// How an object should be treated during an admission when there is a
+	// configuration error preventing CEL evaluation
+	GetFailurePolicy() FailurePolicy
+}
+
+// An interface for internal policy binding type. Implemented by mock/testing types,
+// and to be implemented by the public API types once they have completed API review.
+//
+// The interface closely mirrors the format and functionality of the
+// PolicyBinding proposed in the KEP.
+type PolicyBinding interface {
+	runtime.Object
+
+	// says whether this policy binding matches the provided admission resource request
+	Matches(a admission.Attributes) bool
+
+	// Namespace/Name of Policy Definition used by this binding.
+	GetTargetDefinition() (namespace, name string)
```

Comment:
Do we need namespace returned? What it will be used for?

Reply:
we dont specifically need it, but it doesn't hurt I suppose.

if namespaces don't matter I'd prefer to keep the controller agnostic to this fact and have the definition always just return the same value here, in that case.

Reply:
discussed offline and agreed to have binding `return "", name` if it should be inferred the object is cluster scoped.